bd-report-convert/src/main.rs

```diff
@@ -0,0 +1,68 @@
+// shared-core - bitdrift's common client/server libraries
+// Copyright Bitdrift, Inc. All rights reserved.
+//
+// Use of this source code is governed by a source available license that can be found in the
+// LICENSE file or at:
+// https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt
+
+use std::ffi::{c_char, CString};
+use std::io::Write;
+use std::process::ExitCode;
+use std::{env, fs, slice};
+
+extern "C" {
+  fn bdrr_make_bin_from_json(json_data_path: *const c_char, length_or_err: *mut i32) -> *const u8;
+
+  fn bdrr_print_json(bin_data_path: *const c_char) -> i32;
+}
+
+fn print_usage(program: &str) -> ExitCode {
+  println!("converter for the bd issue report format\n");
+  println!("usage: {program} to-json REPORT_BIN_PATH");
+  println!(
+    "       {} to-bin  REPORT_JSON_PATH DEST_PATH",
+    " ".repeat(program.len())
+  );
+  ExitCode::from(64) // EX_USAGE error
+}
+
+pub fn main() -> Result<ExitCode, std::io::Error> {
+  let args: Vec<String> = env::args().collect();
```

Comment:
you could use `clap` to manage cli arugments but this is simple enough so up to you